azuredevops/internal/service/policy/branch/common.go

```diff
@@ -220,19 +225,37 @@ func expandSettings(d *schema.ResourceData) map[string]interface{} {
 
 		scopeSetting := map[string]interface{}{}
 		if repoID, ok := scopeMap[SchemaRepositoryID]; ok {
-			scopeSetting["repositoryId"] = repoID
+			if repoID == "" {
+				scopeSetting["repositoryId"] = nil
+			} else {
+				scopeSetting["repositoryId"] = repoID
+			}
 		}
 		if repoRef, ok := scopeMap[SchemaRepositoryRef]; ok {
-			scopeSetting["refName"] = repoRef
+			if repoRef == "" {
+				scopeSetting["refName"] = nil
+			} else {
+				scopeSetting["refName"] = repoRef
+			}
 		}
 		if matchType, ok := scopeMap[SchemaMatchType]; ok {
-			scopeSetting["matchKind"] = matchType
+			if matchType == "" {
+				scopeSetting["matchKind"] = nil
+			} else {
+				scopeSetting["matchKind"] = matchType
+			}
+		}
+		if strings.EqualFold(scopeSetting["matchKind"].(string), matchTypeDefaultBranch) && (scopeSetting["repositoryId"] != nil || scopeSetting["refName"] != nil) {
+			return nil, fmt.Errorf("conflicting settings, 'repository_id' is set to %q and 'repository_ref' is set to %q but neither must be set when using a 'match_type' of %q, ",
```

Comment:
`scopeSetting["refName"]` will be `nil` when `matchKind=DefaultBranch`,
```suggestion
			return nil, fmt.Errorf(" neither 'repository_id' nor 'repository_ref' can be set when 'match_type=DefaultBranch'")
```